### PR TITLE
*: Codacy golf

### PR DESCRIPTION
**Pull request by tamird**
Part of testing Codacy's static analysis tools.

### CODE REVIEW

**Comment by arjunravinarayan** [comment]
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/16280)

**Comment by a-robinson** [comment]
Reviewed 1 of 1 files at r1, 1 of 1 files at r2, 1 of 1 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[cloud/kubernetes/init/Dockerfile, line 3 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-Kl_c-0h_eOFxT9yark8:-Kl_c-0h_eOFxT9yark9:b-48kcbn) ([raw file](https://github.com/cockroachdb/cockroach/blob/5abf979283997384a326938b7c6db6574ad2e570/cloud/kubernetes/init/Dockerfile#L3)):*
> ```Dockerfile
> FROM gcr.io/google_containers/peer-finder:0.2
> 
> MAINTAINER Alex Robinson <alexdwanerobinson@gmail.com>
> ```

Do all the different changes to the `cloud/kubernetes/init` directory really deserve separate commits?

---

*[pkg/ccl/storageccl/engineccl/db.cc, line 24 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-Kl_bEjlfK9H1trbhyY1:-Kl_bEjlfK9H1trbhyY2:byzf554) ([raw file](https://github.com/cockroachdb/cockroach/blob/dcd10fed3affca07cb9543bdd3a5617f70a2e8e7/pkg/ccl/storageccl/engineccl/db.cc#L24)):*
> ```cpp
> 
> DBStatus DBBatchReprVerify(
>   DBSlice& repr, DBKey start, DBKey end, int64_t now_nanos, MVCCStatsResult* stats
> ```

Is passing this by reference really better here? This struct only has a pointer and an int in it...

If this would really force us to passing by reference, it should at least be by const reference.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280)*

File: cloud/kubernetes/init/on-start.sh [hunk 1]
```diff
@@ -3,6 +3,6 @@
 # Simply determine if any instances exist other than this one. If there are any
 # others, then assume that a cluster already exists and create a marker to
 # signal that we shouldn't create a new one.
-if grep -v `hostname -f`; then
+if grep -v $(hostname -f); then
```
Comment (tschottdorf) [inline cloud/kubernetes/init/on-start.sh]: needs quotes to be sane, still?

**Comment by bdarnell** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[pkg/ccl/storageccl/engineccl/db.cc, line 24 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-Kl_bEjlfK9H1trbhyY1:-Kl_eRVZa1aZey9HQiXu:bk02l1m) ([raw file](https://github.com/cockroachdb/cockroach/blob/dcd10fed3affca07cb9543bdd3a5617f70a2e8e7/pkg/ccl/storageccl/engineccl/db.cc#L24)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Is passing this by reference really better here? This struct only has a pointer and an int in it...

If this would really force us to passing by reference, it should at least be by const reference.
</blockquote></details>

Yeah. We generally follow [google style](https://google.github.io/styleguide/cppguide.html#Reference_Arguments) and disallow non-const references. 

We pass `DBSlice` by value in a lot of places in the non-CCL `db.cc`, and I'd rather continue doing that than change them all to const references. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280)*

**Comment by tamird** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[cloud/kubernetes/init/Dockerfile, line 3 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-Kl_c-0h_eOFxT9yark8:-Kl_l45AvvHX04CjktOX:b-t10sfa) ([raw file](https://github.com/cockroachdb/cockroach/blob/5abf979283997384a326938b7c6db6574ad2e570/cloud/kubernetes/init/Dockerfile#L3)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Do all the different changes to the `cloud/kubernetes/init` directory really deserve separate commits?
</blockquote></details>

Is it harmful?

---

*[cloud/kubernetes/init/on-start.sh, line 6 at r6](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-Kl_Yw88Bs_Zc6wGuyP7-r6-6:-Kl_mMmYPG16-qw8Xkeg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb4f696b5aae33ffb8c437a56aefbecdd72eb957/cloud/kubernetes/init/on-start.sh#L6)):*
<details><summary><i>Previously, tschottdorf (Tobias Schottdorf) wrote…</i></summary><blockquote>

needs quotes to be sane, still?
</blockquote></details>

Done.

---

*[pkg/ccl/storageccl/engineccl/db.cc, line 24 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-Kl_bEjlfK9H1trbhyY1:-Kl_mUPpP91XlYUnVwSv:bo0c8j1) ([raw file](https://github.com/cockroachdb/cockroach/blob/dcd10fed3affca07cb9543bdd3a5617f70a2e8e7/pkg/ccl/storageccl/engineccl/db.cc#L24)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Yeah. We generally follow [google style](https://google.github.io/styleguide/cppguide.html#Reference_Arguments) and disallow non-const references. 

We pass `DBSlice` by value in a lot of places in the non-CCL `db.cc`, and I'd rather continue doing that than change them all to const references. 
</blockquote></details>

Backed this out.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280)*

**Comment by a-robinson** [comment]
:lgtm:

---

Reviewed 4 of 4 files at r7, 1 of 1 files at r8, 1 of 1 files at r9, 1 of 1 files at r10, 1 of 1 files at r11.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[cloud/kubernetes/init/Dockerfile, line 3 at r5](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-Kl_c-0h_eOFxT9yark8:-KlaLbzgFU0-GmFzY7ex:b1iz8d5) ([raw file](https://github.com/cockroachdb/cockroach/blob/5abf979283997384a326938b7c6db6574ad2e570/cloud/kubernetes/init/Dockerfile#L3)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary><blockquote>

Is it harmful?
</blockquote></details>

I mean it isn't going to hurt anyone, but it would seem to make it more work for someone to see what's changed with the image between releases. Mostly just personal preference, so don't feel you like you need to change it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16280#-:-KlaMOVO_3ZyZ9St0QsV:bnfp4nl)*

**Pull request merged**